TweetBook/Controllers/V1/PostsController.cs

```diff
@@ -1,28 +1,83 @@
 ﻿using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using System.Diagnostics;
 using TweetBook.Contracts.V1;
+using TweetBook.Contracts.V1.Requests;
+using TweetBook.Contracts.V1.Responses;
 using TweetBook.Domain;
+using TweetBook.Services;
 
 namespace TweetBook.Controllers.V1
 { 
     [ApiController]
     public class PostsController : ControllerBase
     {
-        private List<Post>? _posts;
+        private readonly IPostService _postService;
 
-        public PostsController()
+        public PostsController(IPostService postService)
         {
-            _posts = new List<Post>();
-            for (int i = 0; i < 5; i++)
-            {
-                _posts.Add(new Post { Id = Guid.NewGuid().ToString() });
-            }
+            _postService = postService;
         }
 
         [HttpGet(ApiRoutes.Posts.GetAll)]
         public IActionResult GetAll()
         {
-            return Ok(_posts);
+            return Ok(_postService.GetPosts());
+        }
+
+        [HttpGet(ApiRoutes.Posts.Get)]
+        public IActionResult Get([FromRoute] Guid postId)
+        {
+            var post = _postService.GetPostById(postId);
+            if (post == null)
+                return NotFound();
+
+            return Ok(post);
+        }
+
+        [HttpPost(ApiRoutes.Posts.Create)]
+        public IActionResult Create([FromBody] CreatePostRequest postRequest)
+        {
+            var post = new Post { Id = postRequest.Id };
+
+            if (post.Id != Guid.Empty)
+                   post.Id = Guid.NewGuid();
+
+            _postService.GetPosts().Add(post);
+
+            var baseUrl = $"{HttpContext.Request.Scheme}://{HttpContext.Request.Host.ToUriComponent()}";
+            var locationUrl = baseUrl + "/" + ApiRoutes.Posts.Get.Replace("{postId}", post.Id.ToString());

```

Comment:
Same here, use only one type of concatenation

Reply:
I made the changes.